salt/utils/minions.py

```diff
@@ -973,6 +980,65 @@ def auth_check(
             return False
         return False
 
+
+    def check_syndic_auth_list(self, *, minions, funs, args, auth_list):
```

Comment:
this is basically auth_check, but with less pieces. In particular, IIRC, auth_check ends out looking at the PKI dir. We can't do that - we have to compare the minions returned by syndics to their eauth/acl.

So this pulls out all of the code that deals with that, and just compares the functions and commands.